pkg/engine/armresources.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package engine
+
+import (
+	"fmt"
+
+	"github.com/Azure/aks-engine/pkg/api"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func GenerateARMResources(cs *api.ContainerService) []interface{} {
+	var armResources []interface{}
+
+	useManagedIdentity := cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
+	userAssignedIDEnabled := useManagedIdentity && cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID != ""
+
+	if userAssignedIDEnabled {
+		userAssignedID := createUserAssignedIdentities()
+		msiRoleAssignment := createMSIRoleAssignment()
+		armResources = append(armResources, userAssignedID)
```

Comment:
`armResources = append(armResources, userAssignedID, msiRoleAssignment)`

See `// We can add more than one element at a time.` here:

https://tour.golang.org/moretypes/15

Reply:
Sure